### DISCUSSION TITLE
Disable timeout for specific, long endpoints

### DISCUSSION THREAD

**Discussion by ArturFortunato** [fastapi/fastapi#10239] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
router.post("/{project_id:uuid}/video")(ProjectsController.upload_video)

@staticmethod
async def upload_video(
    response: Response,
    project_id: UUID = Path(...),
    video: UploadFile = File(...),
    _: User = Depends(authentication_middleware),
):
   await ProjectService.upload_video(project_id, video)
   ....
```

### Description

I have an endpoint which can very slow, as it allows users to upload randomly large videos. As this operation take very long, it is common for the endpoint to timeout.

How can I disable the timeout JUST for this endpoint? Something like

`router.post("/{project_id:uuid}/video", timeout=0)(ProjectsController.upload_video)`

### Operating System

Linux

### Operating System Details

Running in a `python:3.10.0-slim-buster` Docker container

### FastAPI Version

0.85.0

### Pydantic Version

1.10.2

### Python Version

3.9.6

**Comment by n8sty** [comment, marked as answer]
Where does the timeout come from? If you're using an API gateway product to route traffic to your application that might be the source of the timeout. Same goes for using a proxy like NGINX or Caddy. FastAPI itself does not provide timeout functionality without it being implemented by hand on a project-by-project basis through something like a middleware.

**Reply by danielfcollier** [reply]
The timeout configuration is in the client side. You can add the `httpx` package to write a test for your endpoint and try to catch this behavior:
https://www.python-httpx.org/advanced/#timeout-configuration

If people are uploading videos, it would take a while, so it is good to document this for the users of your API.
